### PR TITLE
feat(Actions): add icon-toggle

### PR DESCRIPTION
**Pull request by talend-glorieux**
**What is the problem this PR is trying to solve?**
We don't have a icon based Action with active or inactive states.

**What is the chosen solution to this problem?**
Adds icon-toggle that follows http://guidelines.talend.com/document/92132#/ui-controls/switch-toggle
(icon size update is in progress in guideline to fit $svg-sm-size)

Depending on the icon, its size should be customizable to have a homogenous visual impression. To enable size (btn and icon) customisation, a sass mixin has been added `@tc-icon-toggle($btn-size, $icon-size)`
See storybook for more details

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/theme/src/theme/_buttons.scss [hunk 1]
```diff
@@ -76,6 +76,43 @@
...
+
+		svg {
+			height: 14px;
+			margin: 4px;
+			width: 14px;
+		}
+
+		&.active {
+			background-color: $brand-primary;
+			border: 1px solid $brand-primary;
```
Comment (jmfrancois) [inline packages/theme/src/theme/_buttons.scss]: border is already set, please change only the border color
unless this is to override existing things ?

File: packages/theme/src/theme/_buttons.scss [hunk 2]
```diff
@@ -76,6 +76,43 @@
 		padding: 0 $padding-smaller;
 	}
 
+	&-active-icon {
+		background-color: unset;
+		border: 1px solid #333;
+		border-radius: 12px;
+		box-shadow: none;
+		color: #333;
```
Comment (jmfrancois) [inline packages/theme/src/theme/_buttons.scss]: where does this color come from ?
We don't have a variable for this one ?

File: packages/theme/src/theme/_buttons.scss [hunk 3]
```diff
@@ -76,6 +76,43 @@
...
+
+		&:hover {
+			opacity: 0.75;
+		}
+
+		svg {
+			height: 14px;
+			margin: 4px;
+			width: 14px;
+		}
```
Comment (jmfrancois) [inline packages/theme/src/theme/_buttons.scss]: please order blocks.
&: first
&.active
svg next

**Comment by jsomsanith** [comment]
`btn-action-icon` --> `.btn-icon` ?

In fact what you designed is a round button, with an icon in it.
So `.btn-icon .active` change the css to have active state

File: packages/theme/src/theme/_buttons.scss [hunk 4]
```diff
@@ -76,6 +76,43 @@
 		padding: 0 $padding-smaller;
 	}
 
+	&-active-icon {
+		background-color: unset;
```
Comment (jsomsanith) [inline packages/theme/src/theme/_buttons.scss]: We should not force the background. If user want to set one of the defined state as default, and activation change it, it won't be possible with that.

File: packages/theme/src/theme/_buttons.scss [hunk 5]
```diff
@@ -76,6 +76,43 @@
...
+		border: 1px solid #333;
+		border-radius: 12px;
+		box-shadow: none;
+		color: #333;
+		height: 24px;
+		line-height: unset;
+		opacity: 0.5;
+		padding: 0;
+		width: 24px;
```
Comment (jsomsanith) [inline packages/theme/src/theme/_buttons.scss]: The width/height/radius are magicnumber. Plus, the button size seems very small compared to the buttons.
Can't we align the sizes with btn's ones ?

File: packages/theme/src/theme/_buttons.scss [hunk 6]
```diff
@@ -76,6 +76,43 @@
...
+		}
+
+		&:hover {
+			opacity: 0.75;
+		}
+
+		svg {
+			height: 14px;
+			margin: 4px;
+			width: 14px;
```
Comment (jsomsanith) [inline packages/theme/src/theme/_buttons.scss]: Use one of the svg predefined size

**Comment by jsomsanith** [comment]
Guideline is in progress

**Comment by jsomsanith** [comment]
http://guidelines.talend.com/document/92132#/ui-controls/switch-toggle

**Comment by jmfrancois** [comment]
we add things which are not bootstrap related here :/

File: packages/theme/src/theme/_variables.scss [hunk 7]
```diff
@@ -195,6 +195,8 @@ $btn-border-radius-large: $border-radius-large !default;
 $btn-border-radius-small: $border-radius-small !default;
 
 $btn-line-height: 3.3rem !default;
+$btn-icon-toggle-size: 24px !default;
```
Comment (frassinier) [inline packages/theme/src/theme/_variables.scss]: there are icon size vars for that

File: packages/theme/src/theme/_variables.scss [hunk 8]
```diff
@@ -195,6 +195,8 @@ $btn-border-radius-large: $border-radius-large !default;
...
+$btn-icon-toggle-icon-size: 14px !default;
```
Comment (frassinier) [inline packages/theme/src/theme/_variables.scss]: And please avoid px

File: packages/theme/src/theme/_visual-helpers.scss [hunk 9]
```diff
@@ -1,6 +1,21 @@
 /// Define button colors based on its background
 /// @access public
 /// @param {Color} $btn-bg [$btn-default-bg] - button background color to tint
```
Comment (frassinier) [inline packages/theme/src/theme/_visual-helpers.scss]: Doc is not up to date?

File: packages/theme/src/theme/_visual-helpers.scss [hunk 10]
```diff
@@ -1,6 +1,21 @@
...
+@mixin btn-icon-toggle($btn-size: $btn-icon-toggle-size, $icon-size: $btn-icon-toggle-icon-size) {
+	height: $btn-size;
+	width: $btn-size;
+	border-radius: $btn-size / 2;
+
+	svg {
+		height: $icon-size;
+		width: $icon-size;
+		margin: ($btn-size - $icon-size - 2 /*(border x 2)*/) / 2;
```
Comment (frassinier) [inline packages/theme/src/theme/_visual-helpers.scss]: Are you confident about svg display?

**Comment by jsomsanith** [comment]
You're right I haven't managed the containers part, I'll add this

**Comment by jsomsanith** [comment]
In fact let's do another PR :)

**Pull request merged**